### PR TITLE
Prevent artifacts in stretched fixed mode

### PR DESCRIPTION
**Pull request by zeruth**

### CODE REVIEW

**Comment by devLotto** [comment]
I'm pretty sure that this will reduce FPS by quite a bit?

**Comment by zeruth** [comment]
Not really at all. It's only called while stretched is enabled, still easily hits 50 frames in performance mode.

**Comment by MESLewis** [comment]
It should be possible to detect when the canvas size has changed and only redraw background then I believe

**Comment by devLotto** [comment]
Try if you can put the code at the end of ``if (lastStretchedDimensions == null || !lastStretchedDimensions.equals(stretchedDimensions))``.

If the client is in fullscreen mode, fillRect takes very long, as it takes a while to fill 2 million pixels.

**Comment by zeruth** [comment]
My bad the import is fixed already, but anyways - that solution isn't good enough because artifacts will remain after user enables plugin. only fixes it on resize.

**Comment by zeruth** [comment]
This is really the solution this problem needs. there is no fps difference even in 4k (besides the AA for non performance mode, which is the culprit itself)

File: runelite-client/src/main/java/net/runelite/client/callback/Hooks.java [hunk 1]
```diff
@@ -283,6 +285,16 @@ public static void draw(MainBufferProvider mainBufferProvider, Graphics graphics
 			stretchedGraphics.drawImage(image, 0, 0, stretchedDimensions.width, stretchedDimensions.height, null);
 
 			image = stretchedImage;
+
+			/*
+				Fill Canvas before drawing stretched image to prevent artifacts.
+			*/
+			if (lastStretchedDimensions == null || !lastStretchedDimensions.equals(stretchedDimensions))
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/callback/Hooks.java]: This should be in the already existing if

**Comment by zeruth** [comment]
Sorry about that, it's moved to the existing statement now.

File: runelite-client/src/main/java/net/runelite/client/callback/Hooks.java [hunk 2]
```diff
@@ -274,6 +275,13 @@ public static void draw(MainBufferProvider mainBufferProvider, Graphics graphics
 				stretchedGraphics = (Graphics2D) stretchedImage.getGraphics();
 
 				lastStretchedDimensions = stretchedDimensions;
+				
+				/*
+					Fill Canvas before drawing stretched image to prevent artifacts.
+				*/
+				graphics.setColor(Color.BLACK);
+				graphics.fillRect(0, 0, client.getCanvas().getWidth(), client.getCanvas().getHeight());
+				graphics.setColor(Color.WHITE);
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/callback/Hooks.java]: Is this setColor unnecessary?

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/callback/Hooks.java]: Hmm, looks like it is.

Comment (zeruth) [reply runelite-client/src/main/java/net/runelite/client/callback/Hooks.java]: The Black setColor is necessary. The white one may or may not be, but I had asked you specifically deathbeam. and you suggested i set it back

Comment (Adam-) [reply runelite-client/src/main/java/net/runelite/client/callback/Hooks.java]: If you remove it... does it still work?

Comment (zeruth) [reply runelite-client/src/main/java/net/runelite/client/callback/Hooks.java]: The black setColor? no, it paints the background white when in fixed resized mode (while not taking up the whole screen) the White one? as my configuration stands yes it works fine, but idk exactly how the graphics hand off will affect any other plugins etc, So I set it back just to be sure.

**Pull request merged**